src/lib.zig

```diff
@@ -3015,6 +3033,251 @@ pub const Lua = struct {
+        const index = lua.absIndex(raw_index);
+        switch (@typeInfo(T)) {
+            .Int => {
+                switch (comptime lang) {
+                    .lua51, .luajit => {
+                        const result = lua.toInteger(index);
+                        return @as(T, @intCast(result));
+                    },
+                    else => {
+                        const result = try lua.toInteger(index);
+                        return @as(T, @intCast(result));
+                    },
+                }
+            },
+            .Float => {
+                switch (comptime lang) {
+                    .lua51, .luajit => {
+                        const result = lua.toNumber(index);
+                        return @as(T, @floatCast(result));
+                    },
+                    else => {
+                        const result = try lua.toNumber(index);
+                        return @as(T, @floatCast(result));
+                    },
+                }
+            },
+            .Pointer => |param_info| {
+                switch (param_info.size) {
+                    .Slice, .Many => {
+                        if (param_info.child != u8) {
+                            @compileError("Only u8 arrays (strings) may be parameters");
+                        }
+                        if (!param_info.is_const) {
+                            @compileError("Slice must be a const slice");
+                        }
+                        const string: [*:0]const u8 = try lua.toString(index);
+                        const end = std.mem.indexOfSentinel(u8, 0, string);
+
+                        if (param_info.sentinel == null) {
+                            return string[0..end];
+                        } else {
+                            return string[0..end :0];
+                        }
+                    },
+                    else => {
+                        return try lua.toUserdata(param_info.child, index);
+                    },
+                }
+            },
+            .Bool => {
+                return lua.toBoolean(index);
+            },
+            .Struct => {
+                return try lua.toStruct(T, index);
+            },
+            .Optional => {
+                if (lua.isNil(index)) {
+                    lua.pop(1);
+                    return null;
+                } else {
+                    return try lua.toAny(@typeInfo(T).Optional.child, index);
+                }
+            },
+            else => {
+                @compileError("Invalid parameter type");
+            },
+        }
+    }
+
+    /// Converts value at given index to a zig struct if possible
+    fn toStruct(lua: *Lua, comptime T: type, raw_index: i32) !T {
+        const index = lua.absIndex(raw_index);
+
+        if (!lua.isTable(index)) {
+            std.log.err("Parsing lua table failed because value at index {} is not a table", .{raw_index});
```

Comment:
I'm admittedly not someone who uses logging very often. But it seems a bit weird to me to be logging from inside a library. What if end users don't want logs?

I think the error returned should be enough to debug

Reply:
From [Andrew Kelly ](https://github.com/ziglang/zig/issues/5738)
 > The goal of the std.log API is to allow developers to leave log statements in finished code, without causing any performance issues or unwanted stderr/stdout for API consumers.
 
 Using parsing api's which fail to tell you why the parsing failed (such as the zig std json library) have caused me a lot of issues in the past. I plan to eventually implement some sort of diagnostics for this. But it seemed using the std.log api was a simple solution. 
 
 End users can set the log level and prevent any logging they don't want.

Reply:
Okay, I think I understand. But at the same time I'm hesitant to add logging for now. Can you remove all of the uses of std.log from the PR and then we can revisit this later?

Reply:
Works for me

Reply:
I made an issue for this: https://github.com/natecraddock/ziglua/issues/59